frontend/tests/features/wheresMyGene.test.ts

```diff
@@ -142,6 +143,69 @@ describe("Where's My Gene", () => {
     }
   });
 
+  test("Source Data", async ({ page }) => {
+    await goToPage(`${TEST_URL}${ROUTES.WHERE_IS_MY_GENE}`, page);
+
+    async function getSourceDataButton() {
+      return page.$(getTestID(SOURCE_DATA_BUTTON_ID));
+    }
+    async function getTissueSelectorButton() {
+      return page.$(getTestID(ADD_TISSUE_ID));
+    }
+
+    async function getGeneSelectorButton() {
+      return page.$(getTestID(ADD_GENE_ID));
+    }
+
+    await clickUntilOptionsShowUp(getTissueSelectorButton, page);
+    await selectFirstOption(page);
+
+    await clickUntilOptionsShowUp(getGeneSelectorButton, page);
+    await selectFirstOption(page);
+
+    await tryUntil(
+      async () => {
+        const canvases = await page.$$("canvas");
+        expect(canvases.length).not.toBe(0);
+      },
+      { page }
+    );
+    await clickUntilSidebarShowsUp(getSourceDataButton, page);
+    await expect(page).toHaveSelector(getText("After filtering cells with low coverage (less than 500 genes expressed)"));
+    const sourceDataList = await page.$("[class*=MuiList-root]");
+    if (!sourceDataList) throw Error("no source data displayed");
+    
+    const sourceDataListItems = await sourceDataList?.$$(".MuiListItem-root")
+    expect(sourceDataListItems?.length).toBe(17);
```

Comment:
Should we just assert that the length is more than 0, since asserting `17` is pretty specific and will break when data changes?